### PR TITLE
Feature/45 common request dispatchig part2

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonString;
+import javax.json.JsonValue;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TablesMetadataParser {
+    private static final String INVOLVED_TABLES = "involvedTables";
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java]: Please move to `RequestParserConstants`.

File: src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    private static final String DATA_TYPE = "dataType";
+
+    public List<TableMetadata> parse(final JsonObject tablesAsJson) throws MetadataException {
+        final List<TableMetadata> tables = new ArrayList<>();
+        final JsonArray jsonArray = tablesAsJson.getJsonArray(INVOLVED_TABLES);
+        for (final JsonObject table : jsonArray.getValuesAs(JsonObject.class)) {
+            final String tableName = table.getString("name", "");
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java]: Extract method. Extract `name` to constant and move it to `RequestParserConstants`.

File: src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            final String tableAdapterNotes = readAdapterNotes(table);
+            final String tableComment = table.getString("comment", "");
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java]: See above.

File: src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        }
+        return tables;
+    }
+
+    private ColumnMetadata parseColumnMetadata(final JsonObject column) {
+        final String columnName = column.getString("name");
+        final String adapterNotes = readAdapterNotes(column);
+        final String comment = column.getString("comment", "");
+        final String defaultValue = column.getString("default", "");
+        final boolean isNullable = checkIfBooleanValueExists(column, "isNullable");
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java]: The name of the method is of. You are not checking the existence as much as you are applying a default value if it does not exist.

File: src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        final boolean isIdentity = checkIfBooleanValueExists(column, "isIdentity");
+        final JsonObject dataType = column.getJsonObject(DATA_TYPE);
+        final DataType type = getDataType(dataType);
+        return new ColumnMetadata(columnName, adapterNotes, type, isNullable, isIdentity, defaultValue, comment);
+    }
+
+    private String readAdapterNotes(final JsonObject root) {
+        if (root.containsKey("adapterNotes")) {
+            final JsonValue notes = root.get("adapterNotes");
+            if (notes.getValueType() == JsonValue.ValueType.STRING) {
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java]: Extract method.

File: src/test/java/com/exasol/adapter/request/parser/TablesMetadataParserTest.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+class TablesMetadataParserTest {
+    @Test
+    void testParseTablesMetadata() throws IOException, MetadataException {
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/request/parser/TablesMetadataParserTest.java]: Method too long split into multiple tests or parameterize test if possible.

File: src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java]: done

File: src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java]: done

File: src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java]: done

File: src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java (hunk 4 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java]: fixed

File: src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java (hunk 5 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java]: fixed

File: src/test/java/com/exasol/adapter/request/parser/TablesMetadataParserTest.java (hunk 6 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/request/parser/TablesMetadataParserTest.java]: fixed

**Pull request merged**